pulumi/infra/swarm.py

```diff
@@ -0,0 +1,328 @@
+            from_port=self.port,
+            to_port=self.port,
+            protocol=self.protocol,
+        )
+        egress = aws.ec2.SecurityGroupEgressArgs(
+            from_port=self.port,
+            to_port=self.port,
+            protocol=self.protocol,
+        )
+        return (ingress, egress)
+
+
+def _attach_AmazonSSMManagedInstanceCore(
+    role: aws.iam.Role, opts: pulumi.ResourceOptions
+) -> aws.iam.RolePolicyAttachment:
+    """
+    https://github.com/grapl-security/issue-tracker/issues/106
+    Moving away from managed server policies.
+    """
+    policy_name = "demanaged_AmazonSSMManagedInstanceCore"
+    policy = aws.iam.Policy(
+        policy_name,
+        policy=json.dumps(
+            {
+                "Version": "2012-10-17",
+                "Statement": [
+                    {
+                        "Effect": "Allow",
+                        "Action": [
+                            "ssm:DescribeAssociation",
+                            "ssm:GetDeployablePatchSnapshotForInstance",
+                            "ssm:GetDocument",
+                            "ssm:DescribeDocument",
+                            "ssm:GetManifest",
+                            "ssm:GetParameter",
+                            "ssm:GetParameters",
+                            "ssm:ListAssociations",
+                            "ssm:ListInstanceAssociations",
+                            "ssm:PutInventory",
+                            "ssm:PutComplianceItems",
+                            "ssm:PutConfigurePackageResult",
+                            "ssm:UpdateAssociationStatus",
+                            "ssm:UpdateInstanceAssociationStatus",
+                            "ssm:UpdateInstanceInformation",
+                        ],
+                        "Resource": "*",
+                    },
+                    {
+                        "Effect": "Allow",
+                        "Action": [
+                            "ssmmessages:CreateControlChannel",
+                            "ssmmessages:CreateDataChannel",
+                            "ssmmessages:OpenControlChannel",
+                            "ssmmessages:OpenDataChannel",
+                        ],
+                        "Resource": "*",
+                    },
+                    {
+                        "Effect": "Allow",
+                        "Action": [
+                            "ec2messages:AcknowledgeMessage",
+                            "ec2messages:DeleteMessage",
+                            "ec2messages:FailMessage",
+                            "ec2messages:GetEndpoint",
+                            "ec2messages:GetMessages",
+                            "ec2messages:SendReply",
+                        ],
+                        "Resource": "*",
+                    },
+                ],
+            }
+        ),
+        opts=opts,
+    )
+    return aws.iam.RolePolicyAttachment(
+        f"attach_{policy_name}",
```

Comment:
(this is me manually attempting to get around an un-needed `apply`; god i'd so much prefer some sort of await type syntax here)